Plugins/AWSLambdaDeployer/Plugin.swift

```diff
@@ -0,0 +1,348 @@
+            print("generating SAM deployment descriptor")
+            configuration.verboseLogging ? print("\(samDeploymentDescriptorUrl)") : nil
+            print("-------------------------------------------------------------------------")
+            let samDeploymentDescriptor = try self.execute(
+                executable: deploymentDescriptorExecutableUrl,
+                arguments: configuration.products.compactMap { $0.name },
+                logLevel: configuration.verboseLogging ? .debug : .silent)
+            try samDeploymentDescriptor.write(
+                to: samDeploymentDescriptorUrl, atomically: true, encoding: .utf8)
+
+            print("-------------------------------------------------------------------------")
+            print("validating SAM deployment descriptor")
+            print("-------------------------------------------------------------------------")
+            try self.execute(
+                executable: samExecutableUrl,
+                arguments: ["validate", "-t", samDeploymentDescriptorUrl.path],
+                logLevel: configuration.verboseLogging ? .debug : .silent)
+
+            if !configuration.noDeploy {
+                print("-------------------------------------------------------------------------")
+                print("deploying AWS Lambda function")
+                print("-------------------------------------------------------------------------")
+                try self.execute(
+                    executable: samExecutableUrl,
+                    arguments: ["deploy", "-t", samDeploymentDescriptorUrl.path],
+                    logLevel: configuration.verboseLogging ? .debug : .silent)
+            }
+        } catch Errors.processFailed(_, _) {
+            print("The generated SAM template is invalid or can not be deployed.")
+            if configuration.verboseLogging {
+                print("File at : \(samDeploymentDescriptorUrl)")
+            } else {
+                print("Run the command again with --verbose argument to receive more details.")
+            }
+        } catch {
+            print("Can not execute file at:")
+            print("\(deploymentDescriptorExecutableUrl.path)")
+            print("or at:")
+            print("\(samExecutableUrl.path)")
+            print("Is SAM installed ? (brew tap aws/tap && brew install aws-sam-cli)")
+            print("Did you add a 'Deploy' executable target into your project's Package.swift ?")
+            print("Did you build the release version ? (swift build -c release)")
+        }
+    }
+
+    @discardableResult
+    private func execute(
+        executable: URL,
+        arguments: [String],
+        customWorkingDirectory: URL? = nil,
+        logLevel: ProcessLogLevel
+    ) throws -> String {
+        try self.execute(
+            executable: Path(executable.path),
+            arguments: arguments,
+            customWorkingDirectory: customWorkingDirectory == nil
```

Comment:
flatMap

Reply:
sorry, I am not sure to understand where you suggest to use `flatMap()` in the 4 lines highlighted above

Reply:
does this `customWorkingDirectory.flatMap { Path($0) }` work?